From 26090b3dbcdf6a11e60535da949b726a6e86426d Mon Sep 17 00:00:00 2001
From: "Maxin B. John" <maxin.john@intel.com>
Date: Tue, 21 Feb 2017 11:49:07 +0200
Subject: [PATCH] configure.ac:
 only-check-conntrack-when-libnfnetlink-enabled.patch

Package libnetfilter-conntrack depends on package libnfnetlink. iptables
checks package libnetfilter-conntrack whatever its package config
libnfnetlink is enabled or not. When libnfnetlink is disabled but
package libnetfilter-conntrack exists, it fails randomly with:

In file included from
.../iptables/1.4.21-r0/iptables-1.4.21/extensions/libxt_connlabel.c:8:0:

.../tmp/sysroots/qemumips/usr/include/libnetfilter_conntrack/libnetfilter_conntrack.h:14:42:
fatal error: libnfnetlink/linux_nfnetlink.h: No such file or directory

compilation terminated.
GNUmakefile:96: recipe for target 'libxt_connlabel.oo' failed
Only check libnetfilter-conntrack when libnfnetlink is enabled to fix it.

Upstream-Status: Pending

Signed-off-by: Kai Kang <kai.kang@windriver.com>
Signed-off-by: Maxin B. John <maxin.john@intel.com>
---
 configure.ac | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/configure.ac b/configure.ac
index 03ddc50..523caea 100644
--- a/configure.ac
+++ b/configure.ac
@@ -172,10 +172,12 @@ if test "$nftables" != 1; then
 fi
 
 if test "x$enable_connlabel" = "xyes"; then
-	PKG_CHECK_MODULES([libnetfilter_conntrack],
+    nfconntrack=0
+    AS_IF([test "x$enable_libnfnetlink" = "xyes"], [
+    PKG_CHECK_MODULES([libnetfilter_conntrack],
 		[libnetfilter_conntrack >= 1.0.6],
 		[nfconntrack=1], [nfconntrack=0])
-
+    ])
 	if test "$nfconntrack" -ne 1; then
 		blacklist_modules="$blacklist_modules connlabel";
 		echo "WARNING: libnetfilter_conntrack not found, connlabel match will not be built";
-- 
2.4.0

